paasta_tools/tron_tools.py

```diff
@@ -286,123 +284,91 @@ def __init__(
+            "spark.app.name",
+            f"tron_spark_{self.get_service()}_{self.get_instance()}",
         )
 
-        # Core ML has some translation logic to go from Mesos->k8s options for people living in the past
-        conf.update(
-            adjust_spark_resources(
-                spark_args=stringified_spark_args, desired_pool=self.get_pool()
+        docker_img_url = self.get_docker_url(system_paasta_config)
+
+        spark_conf_builder = SparkConfBuilder()
+        spark_conf = spark_conf_builder.get_spark_conf(
+            cluster_manager="kubernetes",
+            spark_app_base_name=spark_app_name,
+            user_spark_opts=stringified_spark_args,
+            paasta_cluster=resolved_cluster,
+            paasta_pool=self.get_spark_executor_pool(),
+            paasta_service=self.get_service(),
+            paasta_instance=self.get_instance(),
+            docker_img=docker_img_url,
+            extra_volumes=self.get_volumes(system_paasta_config.get_volumes()),
+            use_eks=True,
+            k8s_server_address=get_k8s_url_for_cluster(self.get_cluster()),
+            force_spark_resource_configs=self.config_dict.get(
+                "force_spark_resource_configs", False
+            ),
+            user=spark_tools.SPARK_JOB_USER,
+        )
+        # TODO: Remove this once dynamic pod template is generated inside the driver using spark-submit wrapper
+        if "spark.kubernetes.executor.podTemplateFile" in spark_conf:
+            print(
+                f"Replacing spark.kubernetes.executor.podTemplateFile="
+                f"{spark_conf['spark.kubernetes.executor.podTemplateFile']} with "
+                f"spark.kubernetes.executor.podTemplateFile={spark_tools.SPARK_DNS_POD_TEMPLATE}"
             )
+            spark_conf[
+                "spark.kubernetes.executor.podTemplateFile"
+            ] = spark_tools.SPARK_DNS_POD_TEMPLATE
+
+        spark_conf.update(
+            {
+                "spark.hadoop.fs.s3a.aws.credentials.provider": spark_tools.SPARK_AWS_CREDS_PROVIDER,
+                "spark.driver.host": "$PAASTA_POD_IP",
+            }
+        )
+        spark_conf.setdefault(
+            "spark.kubernetes.executor.label.yelp.com/owner", self.get_team()
         )
-        # as well as some QoL tweaks for other options
-        conf.update(setup_shuffle_partitions(spark_args=stringified_spark_args))
-        conf.update(setup_event_log_configuration(spark_args=stringified_spark_args))
 
-        return conf
+        # We need to make sure the Service Account used by the executors has been created.
+        # We are using the Service Account created using the provided or default IAM role.
+        spark_conf[
+            "spark.kubernetes.authenticate.executor.serviceAccountName"
+        ] = create_or_find_service_account_name(
+            iam_role=self.get_spark_executor_iam_role(),
+            namespace=spark_tools.SPARK_EXECUTOR_NAMESPACE,
```

Comment:
btw "SPARK_EXECUTOR_NAMESPACE"  in  spark_tools file is set to ``paasta-spark`` namespace, is that intended? and I think we are using the tron namespace in "EXECUTOR_TYPE_TO_NAMESPACE"

Reply:
or is it that we want to run the driver in the tron namespace and the executor in paasta-spark ns?

Reply:
Okay I see tron will launch the drivers on the tron ns and the executor will be ran on the paasta-spark ns